src/blockdata/transaction.rs

```diff
@@ -111,14 +111,22 @@ pub enum TxOutTarget {
 }
 
 impl TxOutTarget {
-    /// Retreive the public keys, if any.
+    /// Retrieve the public keys, if any.
     pub fn get_pubkeys(&self) -> Option<Vec<PublicKey>> {
         match self {
             TxOutTarget::ToScript { keys, .. } => Some(keys.clone()),
             TxOutTarget::ToKey { key } => Some(vec![*key]),
             TxOutTarget::ToScriptHash { .. } => None,
         }
     }
+
+    /// Returns the one-time public key if this is a [`TxOutTarget::ToKey`] and `None` otherwise.
+    pub fn as_to_key(&self) -> Option<&PublicKey> {
```

Comment:
:wrench: This name is confusing because of Rust naming conventions. Perhaps we could say `as_one_time_key` or `as_one_time_pk`.